packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts

```diff
@@ -101,8 +103,41 @@ describe('query preview', () => {
     expect(getXComponentXModuleName(wrapper.vm)).toBe('queriesPreview');
   });
 
+  // eslint-disable-next-line max-len
+  it('does not send the `QueryPreviewRequestUpdated` event if persistInCache is true, but emits load', () => {
+    const { queryPreviewRequestUpdatedSpy, wrapper } = renderQueryPreview({
+      persistInCache: true,
+      queryPreviewInfo: {
+        query: 'shoes',
+        extraParams: { directory: 'Magrathea' },
+        filters: ['fit:regular']
+      }
+    });
+
+    jest.advanceTimersByTime(0); // Wait for first emission.
+    expect(queryPreviewRequestUpdatedSpy).toHaveBeenCalledTimes(0);
+    expect(wrapper.emitted('load')?.length).toBe(1);
+    expect(wrapper.emitted('load')?.[0]).toEqual(['shoes']);
+  });
+
+  it('does not remove the query before destroying the component', () => {
```

Comment:
We're still not testing well the functionality. We have to test both cases, `persistInCache=true` and `persistInCache=false`